core/storage/email/gae_models.py

```diff
@@ -184,6 +185,40 @@ def _pre_put_hook(self):
         self.email_hash = self._generate_hash(
             self.recipient_id, self.subject, self.html_body)
 
+    @classmethod
+    def create_multi(cls, send_email_infos):
+        """Creates SentEmailModel entries for multiple emails.
+
+        Args:
+            send_email_infos: list(SendEmailInfo). Each SendEmailInfo object
+                contains the information needed to generate a SentEmailModel.
+        """
+        email_model_instances = []
+
+        for send_email_info in send_email_infos:
+            instance_id = cls._generate_id(send_email_info.intent)
+            email_hash = cls._generate_hash(
+                send_email_info.recipient_id, send_email_info.email_subject,
+                send_email_info.email_html_body)
+
+            email_model_instance = cls(
+                id=instance_id, recipient_id=send_email_info.recipient_id,
+                recipient_email=send_email_info.recipient_email,
+                sender_id=send_email_info.sender_id,
+                # Sender emails are stored in the following format.
+                sender_email='%s <%s>' % (
```

Comment:
Why do this & the sent_datetime fields deviate from create()? Shouldn't create_multi() behave in the same way as create(), e.g. requiring all information to be passed in through the parameters?

Reply:
I actually think we should also be defining the datetime at the domain object level or higher--in general, models shouldn't fix the values of fields except for default cases otherwise this can make it difficult to get a complete picture of what is stored in a given model at any given time (another way to think about this: prefer to decide what to store in a model in one location).

Reply:
N/A anymore since this was removed